### PR TITLE
Update README and iterator docs

### PR DESCRIPTION
**Pull request by JanuszL**
- adds more hints on how to build DALI on bare metal - how to set paths
  to some libs
- adds info on how to set DALI build type
- adds a more verbose description of `size` in Pytorch and MXNet iterators

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

File: dali/python/nvidia/dali/plugin/mxnet.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ class DALIGenericIterator(object):
                  for data or label correspondingly.
                  output_names should be distinct.
     size : int
-           Epoch size.
+          Number of samples in the epoch in the whole data set.
```
Comment (Kh4L) [inline dali/python/nvidia/dali/plugin/mxnet.py]: > in the whole data set.
Why? This is what user usually wants to do but you can set it to anything else (e.g. for testing).

Comment (Kh4L) [inline dali/python/nvidia/dali/plugin/mxnet.py]: What about something like
```suggestion
          Number of samples in the epoch (Usually the size of the dataset).
```

File: dali/python/nvidia/dali/plugin/pytorch.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ class DALIGenericIterator(object):
                  of those names.
                  Each name should be distinct
     size : int
-           Epoch size.
+           Number of samples in the epoch in the whole data set.
```
Comment (Kh4L) [inline dali/python/nvidia/dali/plugin/pytorch.py]: Same here

File: dali/python/nvidia/dali/plugin/mxnet.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/mxnet.py]: Done

Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/mxnet.py]: Done

File: dali/python/nvidia/dali/plugin/pytorch.py (hunk 2 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/pytorch.py]: Done

**Pull request merged**